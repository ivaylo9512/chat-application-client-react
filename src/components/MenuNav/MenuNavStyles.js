import styled from 'styled-components'

export const Nav = styled.nav`
    position: relative;
    height: 32vw;
    width: 32vw;

    li:first-child {
        transform: rotate(-5deg);
    }
    li:nth-child(2) {
        transform: rotate(15deg);
    }
    li:nth-child(3) {
        transform: rotate(35deg);
    }
    li:nth-child(4) {
        transform: rotate(55deg);
    }
    li:nth-child(5) {
        transform: rotate(75deg);
    }
    li:nth-child(6) {
        transform: rotate(95deg);
    }
    li:nth-child(6) {
        transform: rotate(95deg);
    }
    li:nth-child(6) {
        transform: rotate(95deg);
    }
    li:nth-child(7) {
        transform: rotate(120deg);
    }
    li:nth-child(8) {
        transform: rotate(140deg);
    }
    li:nth-child(9) {
        transform: rotate(160deg);
    }
`

export const Li = styled.li`
    position: absolute;
    font-size: 1.5em;
    width: 4vw;
    height: 100%;
    left: 50%;
    margin-left: -2vw;
    transform-origin: center;
    list-style: none;

    a{
        text-align: center;
        line-height: 3vw;
        position: absolute;
        top: 1vw;
        border-radius: 50%;
        font-size: 0.95vw;
        color: white;
        height: 3vw;
        width: 3vw;
        background: rgb(24,27, 33);
        padding: 0;
        border: 0;
        box-shadow: 0px 0px 0.5vw 0.1vw rgb(0, 0, 0, 0.8)
    }
`

export const Button = styled.button`
    position: absolute;
    top: 1vw;
    border-radius: 50%;
    font-size: 0.95vw;
    color: white;
    height: 3vw;
    width: 3vw;
    background: rgb(24,27, 33);
    padding: 0;
    border: 0;
    box-shadow: 0px 0px 0.5vw 0.1vw rgb(0, 0, 0, 0.8);
`

export const Span = styled.span`
    font-size: 1.1em;
    opacity: 0.7;
`