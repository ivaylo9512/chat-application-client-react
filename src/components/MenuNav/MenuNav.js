import React from 'react'
import { Link, useLocation } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { getStylesState, toggleHeaderVisibility, toggleSearchVisibility } from 'app/slices/stylesSlice';
import { Button, Li, Nav } from './MenuNavStyles'

const MenuNav = ({rotate}) => {
    const location = useLocation();
    const {isHeaderHidden, isSearchHidden} = useSelector(getStylesState)
    const dispatch = useDispatch();

    const toggleHeader = () => { 
        dispatch(toggleHeaderVisibility());
    }
    
    const toggleSearch = () => {
        dispatch(toggleSearchVisibility());
    }

    return(
        <Nav style={rotate}>
            <ul>
                <Li>
                    <Button tabIndex='-1'>
                        <i className='fas fa-user'></i>
                    </Button>
                </Li>
                <Li>
                    <Button onClick={toggleHeader} tabIndex='-1'>
                        <i className={isHeaderHidden  ? 'far fa-eye' : 'far fa-eye-slash'}></i>
                    </Button>
                </Li>
                <Li>
                    <Link to='/searchChat' tabIndex='-1'>
                        <i className='fas fa-search'></i>
                    </Link>
                </Li>
                <Li>
                    <Link to='/searchUsers' tabIndex='-1'>
                        <i className='fas fa-user-plus'></i>
                    </Link>
                </Li>
                {(location.pathname == '/searchUsers' || location.pathname == '/searchChat') &&
                    <Li>
                        <Button onClick={toggleSearch}>
                            <i className={isSearchHidden ? 'fas fa-caret-up' : 'fas fa-caret-down' }></i>
                        </Button>
                    </Li>
                }
                <Li>
                    <Link to='/logout' tabIndex='-1'>
                        <i className='fas fa-sign-out-alt'></i>
                    </Link>
                </Li>
            </ul>
        </Nav>
    )
}
export default MenuNav