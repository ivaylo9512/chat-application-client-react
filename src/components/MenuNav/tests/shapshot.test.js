import MenuNav from 'components/MenuNav/MenuNav';
import { mount } from 'enzyme';
import * as Redux from 'react-redux';
import { Router, MemoryRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';

describe('MenuNav snapshot tests', () => {
    let selectorSpy;

    beforeAll(() => {
        selectorSpy = jest.spyOn(Redux, 'useSelector');
    
        const spyOnUseDispatch = jest.spyOn(Redux, 'useDispatch');
        spyOnUseDispatch.mockReturnValue(jest.fn());
    });

    const createWrapper = (state, path) => {
        selectorSpy.mockReturnValue(state);
    
        return mount(
            <MemoryRouter initialEntries={[path]}>
                <MenuNav />
            </MemoryRouter>
            
        )
    }

    it('should match snapshot when header is not hidden', () => {
        const wrapper = createWrapper({isHeaderHidden: false, isSearchHidden: false}, '/');

        expect(wrapper.find(MenuNav)).toMatchSnapshot();
    })
    
    it('should match snapshot when search is not hidden and path is /searchChat', () => {
        const wrapper = createWrapper({isHeaderHidden: false, isSearchHidden: false}, '/searchChat');

        expect(wrapper.find(MenuNav)).toMatchSnapshot();
    })

    it('should match snapshot when search is hidden and path is /searchChat', () => {
        const wrapper = createWrapper({isHeaderHidden: false, isSearchHidden: true}, '/searchChat');

        expect(wrapper.find(MenuNav)).toMatchSnapshot();
    })

    it('should match snapshot when header is hidden', () => {
        const wrapper = createWrapper({isHeaderHidden: true, isSearchHidden: false}, '/');

        expect(wrapper.find(MenuNav)).toMatchSnapshot();
    })
})