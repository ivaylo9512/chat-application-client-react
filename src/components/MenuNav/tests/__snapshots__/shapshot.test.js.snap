// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuNav snapshot tests should match snapshot when header is hidden 1`] = `
<MenuNav>
  <styled.nav>
    <nav
      className="sc-bdnxRM dCmWw"
    >
      <ul>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              tabIndex="-1"
            >
              <button
                className="sc-dlnjwi hodYHm"
                tabIndex="-1"
              >
                <i
                  className="fas fa-user"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              data-testid="toggleHeader"
              onClick={[Function]}
              tabIndex="-1"
            >
              <button
                className="sc-dlnjwi hodYHm"
                data-testid="toggleHeader"
                onClick={[Function]}
                tabIndex="-1"
              >
                <i
                  className="far fa-eye"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="searchChat"
              tabIndex="-1"
              to="/searchChat"
            >
              <LinkAnchor
                data-testid="searchChat"
                href="/searchChat"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="searchChat"
                  href="/searchChat"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-search"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="searchUsers"
              tabIndex="-1"
              to="/searchUsers"
            >
              <LinkAnchor
                data-testid="searchUsers"
                href="/searchUsers"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="searchUsers"
                  href="/searchUsers"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-user-plus"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="requests"
              tabIndex="-1"
              to="/requests"
            >
              <LinkAnchor
                data-testid="requests"
                href="/requests"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="requests"
                  href="/requests"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-address-card"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="logout"
              tabIndex="-1"
              to="/logout"
            >
              <LinkAnchor
                data-testid="logout"
                href="/logout"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="logout"
                  href="/logout"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-sign-out-alt"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
      </ul>
    </nav>
  </styled.nav>
</MenuNav>
`;

exports[`MenuNav snapshot tests should match snapshot when header is not hidden 1`] = `
<MenuNav>
  <styled.nav>
    <nav
      className="sc-bdnxRM dCmWw"
    >
      <ul>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              tabIndex="-1"
            >
              <button
                className="sc-dlnjwi hodYHm"
                tabIndex="-1"
              >
                <i
                  className="fas fa-user"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              data-testid="toggleHeader"
              onClick={[Function]}
              tabIndex="-1"
            >
              <button
                className="sc-dlnjwi hodYHm"
                data-testid="toggleHeader"
                onClick={[Function]}
                tabIndex="-1"
              >
                <i
                  className="far fa-eye-slash"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="searchChat"
              tabIndex="-1"
              to="/searchChat"
            >
              <LinkAnchor
                data-testid="searchChat"
                href="/searchChat"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="searchChat"
                  href="/searchChat"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-search"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="searchUsers"
              tabIndex="-1"
              to="/searchUsers"
            >
              <LinkAnchor
                data-testid="searchUsers"
                href="/searchUsers"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="searchUsers"
                  href="/searchUsers"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-user-plus"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="requests"
              tabIndex="-1"
              to="/requests"
            >
              <LinkAnchor
                data-testid="requests"
                href="/requests"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="requests"
                  href="/requests"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-address-card"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="logout"
              tabIndex="-1"
              to="/logout"
            >
              <LinkAnchor
                data-testid="logout"
                href="/logout"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="logout"
                  href="/logout"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-sign-out-alt"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
      </ul>
    </nav>
  </styled.nav>
</MenuNav>
`;

exports[`MenuNav snapshot tests should match snapshot when search is hidden and path is /searchChat 1`] = `
<MenuNav>
  <styled.nav>
    <nav
      className="sc-bdnxRM dCmWw"
    >
      <ul>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              tabIndex="-1"
            >
              <button
                className="sc-dlnjwi hodYHm"
                tabIndex="-1"
              >
                <i
                  className="fas fa-user"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              data-testid="toggleHeader"
              onClick={[Function]}
              tabIndex="-1"
            >
              <button
                className="sc-dlnjwi hodYHm"
                data-testid="toggleHeader"
                onClick={[Function]}
                tabIndex="-1"
              >
                <i
                  className="far fa-eye-slash"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="searchChat"
              tabIndex="-1"
              to="/searchChat"
            >
              <LinkAnchor
                data-testid="searchChat"
                href="/searchChat"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="searchChat"
                  href="/searchChat"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-search"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="searchUsers"
              tabIndex="-1"
              to="/searchUsers"
            >
              <LinkAnchor
                data-testid="searchUsers"
                href="/searchUsers"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="searchUsers"
                  href="/searchUsers"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-user-plus"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="requests"
              tabIndex="-1"
              to="/requests"
            >
              <LinkAnchor
                data-testid="requests"
                href="/requests"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="requests"
                  href="/requests"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-address-card"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              data-testid="toggleSearch"
              onClick={[Function]}
            >
              <button
                className="sc-dlnjwi hodYHm"
                data-testid="toggleSearch"
                onClick={[Function]}
              >
                <i
                  className="fas fa-caret-up"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="logout"
              tabIndex="-1"
              to="/logout"
            >
              <LinkAnchor
                data-testid="logout"
                href="/logout"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="logout"
                  href="/logout"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-sign-out-alt"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
      </ul>
    </nav>
  </styled.nav>
</MenuNav>
`;

exports[`MenuNav snapshot tests should match snapshot when search is not hidden and path is /searchChat 1`] = `
<MenuNav>
  <styled.nav>
    <nav
      className="sc-bdnxRM dCmWw"
    >
      <ul>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              tabIndex="-1"
            >
              <button
                className="sc-dlnjwi hodYHm"
                tabIndex="-1"
              >
                <i
                  className="fas fa-user"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              data-testid="toggleHeader"
              onClick={[Function]}
              tabIndex="-1"
            >
              <button
                className="sc-dlnjwi hodYHm"
                data-testid="toggleHeader"
                onClick={[Function]}
                tabIndex="-1"
              >
                <i
                  className="far fa-eye-slash"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="searchChat"
              tabIndex="-1"
              to="/searchChat"
            >
              <LinkAnchor
                data-testid="searchChat"
                href="/searchChat"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="searchChat"
                  href="/searchChat"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-search"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="searchUsers"
              tabIndex="-1"
              to="/searchUsers"
            >
              <LinkAnchor
                data-testid="searchUsers"
                href="/searchUsers"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="searchUsers"
                  href="/searchUsers"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-user-plus"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="requests"
              tabIndex="-1"
              to="/requests"
            >
              <LinkAnchor
                data-testid="requests"
                href="/requests"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="requests"
                  href="/requests"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-address-card"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <styled.button
              data-testid="toggleSearch"
              onClick={[Function]}
            >
              <button
                className="sc-dlnjwi hodYHm"
                data-testid="toggleSearch"
                onClick={[Function]}
              >
                <i
                  className="fas fa-caret-down"
                />
              </button>
            </styled.button>
          </li>
        </styled.li>
        <styled.li>
          <li
            className="sc-gtsrHT cBCFnl"
          >
            <Link
              data-testid="logout"
              tabIndex="-1"
              to="/logout"
            >
              <LinkAnchor
                data-testid="logout"
                href="/logout"
                navigate={[Function]}
                tabIndex="-1"
              >
                <a
                  data-testid="logout"
                  href="/logout"
                  onClick={[Function]}
                  tabIndex="-1"
                >
                  <i
                    className="fas fa-sign-out-alt"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </styled.li>
      </ul>
    </nav>
  </styled.nav>
</MenuNav>
`;
